### ISSUE TITLE
hb-service on MacOS

### ISSUE THREAD

**Issue by UKenGB** [oznu/homebridge-config-ui-x#1386]
### Current Situation

Been running Homebridge since before hb-service was available, so Homebridge is run from a LaunchDaemon plist. However hb-service is apparently now preferred, but instructions to 'Swap from standalone to hb-service' are not aimed at MacOS.

So what is best way to 'update' to hb-service and how does that handle what launchd and the plist do currently? Does hb-service still actually use the existing LaunchDaemon plist? Create a new one? If the latter, how does it keep Homebridge running and in any case, how is sudo not then required?

Basically some explanation of hb-service and how it works on a Mac would be much appreciated.

### Homebridge UI Version

4.49

### Homebridge Version

1.5

### Node.js Version

16.16

### Operating System

macOS

### Environment Info

- [ ] Using Docker?
- [ ] Using Hyper-V?
- [ ] Using hb-service?

**Comment by UKenGB** [comment]
What do you mean by 'system service'? All LaunchDaemons are loaded at startup. Only LaunchAgents require a user to log in. So whether the plist is in /System/Library/LaunchDaemons or /Library/LaunchDaemons, it will be loaded/run at startup, before any user needs to log in.

However, no third party should be chucking plists in /System/Library/LaunchDaemons as that is reserved for basic system essentials and additional 'services' such as HomeBridge should be run from /Library/LaunchDaemons.

So I'm unclear how hb-service runs and controls Homebridge. Does it use launchd? Where is the plist placed?

I like the idea of hb-service, but need to know more about what it's doing to understand how to successfully migrate.

**Comment by UKenGB** [comment]
Ah yes, I just found some additional info on this. I already have the LaunchDaemon plist. I might start again though and let hb-service create a new one.

Reason for reconfiguring is changing the user running Homebridge and trying to tidy it all up as somehow npm had vanished (some time ago) and I couldn't update anything then a few days ago it all just stopped working. Home.app had no idea whether devices etc were on or off, neither had Homebridge, it was all a mess. With npm re-installed and everything updated, it seems to be working again, but would like to ensure hb-service is able to do its stuff.

Thanks.


Ken  G i l l e t t

_/_/_/_/_/_/_/_/

**Comment by UKenGB** [comment]
All done and working. However, could I suggest you add a 'status' command to hb-service?

At the commandline it would be useful to be able to quickly see if homebridge is running and if so, some basic stats like for how long it's been up etc, much as shown in the web UI. Or of course just announcing it's not running.

I would certainly like to see that added.


Ken  G i l l e t t

_/_/_/_/_/_/_/_/
